driver-bitfury16.c

```diff
@@ -0,0 +1,4589 @@
+			gen_clock_data(bf16_renonce_chip_clock, 1, clock_buf);
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].clock = bf16_renonce_chip_clock;
+		} else {
+			gen_clock_data(bf16_chip_clock, 1, clock_buf);
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].clock = bf16_chip_clock;
+		}
+
+		ret = cmd_buffer_push(cmd_buffer,
+				info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+				work, 0, CHIP_CMD_SET_CLOCK, 3, clock_buf);
+
+		if (ret < 0)
+			applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d], error prepare set_clock_cmd",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id);
+		break;
+
+	/* fill chip buffer with chip mask */
+	case CLOCK_SET:
+		applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], prepare set_mask_cmd",
+				bitfury->drv->name,
+				board_id, bcm250_id, chip_id);
+
+		uint8_t noncemask[4];
+		memset(noncemask, 0, sizeof(noncemask));
+
+		for (i = 0; i < 4; i++)
+			noncemask[i] = (mask >> (8*(4 - i - 1))) & 0xff;
+
+		ret = cmd_buffer_push(cmd_buffer,
+				info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+				work, 0, CHIP_CMD_SET_MASK, 3, noncemask);
+
+		if (ret < 0)
+			applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d], error prepare set_mask_cmd",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id);
+		break;
+
+	/* fill chip buffer with new task */
+	case MASK_SET:
+		if ((renonce_chip(chip_address) == 0) ||
+			(opt_bf16_renonce == RENONCE_DISABLED)) {
+			applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], prepare send_task_cmd",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id);
+
+			L_LOCK(info->work_list);
+			L_LOCK(info->stale_work_list);
+			if (info->work_list->count > 0) {
+				memset(info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task, 0,
+						sizeof(info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task));
+
+				bf_data_t* wdata = info->work_list->head;
+
+				workd_list_push(info->stale_work_list, WORKD(wdata));
+				workd_list_remove(info->work_list, bitfury, &info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork);
+
+				gen_task_data(info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.midstate,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.m7,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.ntime,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.nbits, mask,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task);
+
+				ret = cmd_buffer_push(cmd_buffer,
+						info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork, 0,
+						CHIP_CMD_TASK_WRITE, 79,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task);
+
+				if (ret < 0)
```

Comment:
driver-bitfury16.c:1583:13: warning: comparison is always false due to limited range of data type [-Wtype-limits]
     if (ret < 0)